EasyReflectometryApp/Logic/Proxies/Project.py

```diff
@@ -112,213 +92,38 @@ def currentProjectPath(self, new_path):
 
     @Slot()
     def createProject(self):
-        projectPath = self.currentProjectPath
-        mainCif = os.path.join(projectPath, 'project.cif')
-        samplesPath = os.path.join(projectPath, 'samples')
-        experimentsPath = os.path.join(projectPath, 'experiments')
-        calculationsPath = os.path.join(projectPath, 'calculations')
-        if not os.path.exists(projectPath):
-            os.makedirs(projectPath)
-            os.makedirs(samplesPath)
-            os.makedirs(experimentsPath)
-            os.makedirs(calculationsPath)
-            with open(mainCif, 'w') as file:
-                file.write(self.projectInfoAsCif)
-        else:
-            print(f"ERROR: Directory {projectPath} already exists")
+        self._project.create_project_dir()
+        self._project.default_model()
+        self._relayProjectChange()
+        self.saveProject()
 
-    @Slot()
-    def saveProject(self):
-        self._saveProject()
+    def _relayProjectChange(self):
+        self.parent._model_proxy._model = self._project._models
+        self.parent._material_proxy._materials = self._project._materials
+        self._project_created = True
+
+        self.projectCreatedChanged.emit()
+        self.parent.sampleChanged.emit()
         self.parent._state_proxy.stateChanged.emit(False)
 
     @Slot(str)
     def loadProjectAs(self, filepath):
-        self._loadProjectAs(filepath)
-        self.parent._state_proxy.stateChanged.emit(False)
+        self._project.load_project_json(generalizePath(filepath))
+        self._relayProjectChange()
 
     @Slot()
     def loadProject(self):
-        self._loadProject()
+        self._project.load_project_json()
+        self._relayProjectChange()
+
+    @Slot()
+    def saveProject(self):
+        self._project.save_project_json()
         self.parent._state_proxy.stateChanged.emit(False)
```

Comment:
This is an old code, but please also consider not calling `_state_proxy` directly.
Could be done as a separate maintenance fix.